docs/proposals/20221019-capi-managed-cluster-proposal-capz.md

```diff
@@ -0,0 +1,185 @@
+
+     // NetworkSpec encapsulates all things related to the GCP network.
+     // +optional
+     Network NetworkSpec `json:"network"`
+
+     // FailureDomains is an optional field which is used to assign selected availability zones to a cluster
+     // FailureDomains if empty, defaults to all the zones in the selected region and if specified would override
+     // the default zones.
+     // +optional
+     FailureDomains []string `json:"failureDomains,omitempty"`
+
+     // AdditionalLabels is an optional set of tags to add to GCP resources managed by the GCP provider, in addition to the
+     // ones added by default.
+     // +optional
+     AdditionalLabels Labels `json:"additionalLabels,omitempty"`
+
+     ...
+ }
+```
+
+The option above is the best way to proceed for **new implementations** of Managed Kubernetes in a provider.
+
+ The reasons for this recommendation are as follows:
+
+ - It adheres closely to the original separation of concerns between the infra and control plane providers
+ - The infra cluster provisions and manages the general infrastructure required for the cluster but not the control plane.
+ - By having a separate infra cluster API definition, it allows differences in the API between managed and unmanaged clusters.
+
+ Providers like CAPZ and CAPA have already implemented Managed Kubernetes support and there should be no requirement on them to move to Option 3. Both Options 2 and 4 are solutions that would work with ClusterClass and so could be used if required.
+
+ Option 1 is the only option that will not work with ClusterClass and would require a change to CAPI. Therefore this option is not recommended.
+
+ *** This means that CAPA will have to make changes to move away from Option 1 if it wants to support ClusterClass.
+
+# CAPZ Ramifications
+
+CAPZ's current Managed Kubernetes solution fulfills part of the above recommendation: it implements distinct `AzureManagedCluster` and `AzureManagedControlPlane` CRDs, in compliance with the Cluster API `ClusterClass` specification.
+
+It differs from the proposal in the following ways:
+
+- `AzureManagedCluster` does not exclusively define "cluster"-specific properties (the examples above are networking and IAM)
+- `AzureManagedControlPlane` does not exclusively define "control plane"-specific properties; in fact it defines the entire set of "cluster"-wide configuration supported by CAPZ on top of AKS
+
+In order to better adhere to the recommended implementation model CAPZ would move the bulk of its configuration surface area from `AzureManagedControlPlane` to `AzureManagedCluster`.
+
+## AKS-CAPI API Affinity Observations
+
+As stated above, Cluster API defines a "cluster" as distinct from a "control plane". AKS, however, does not declare such a clean boundary in its own API. Using the `az aks` CLI as an example, we can see that the only current definite abstraction boundary at present (as distinct from the cluster itself is "nodepool". E.g.:
```

Comment:
Thanks for explaining it with examples. When we wrote the initial managed k8s proposal, we also had difficulties placing some fields into two separate CRDs as it was not very clear whether they belonged to control plane or cluster. Great to understand the perspective from AKS architecture better.